be/src/exec/pipeline/crossjoin/cross_join_context.cpp

```diff
@@ -43,4 +45,67 @@ Status CrossJoinContext::_init_runtime_filter(RuntimeState* state) {
     }
     return Status::OK();
 }
+
+bool CrossJoinContext::finish_probe(int32_t driver_seq, const std::vector<uint8_t>& build_match_flags) {
+    std::lock_guard guard(_join_stage_mutex);
+
+    ++_num_post_probers;
+    VLOG(3) << fmt::format("CrossJoin operator {} finish probe {}/{}: {}", driver_seq, _num_post_probers,
+                           _num_left_probers, fmt::join(build_match_flags, ","));
+    bool is_last = _num_post_probers == _num_left_probers;
+
+    // Merge all build_match_flag from all probers
+    if (build_match_flags.empty()) {
+        return is_last;
+    }
+    if (_shared_build_match_flag.empty()) {
+        _shared_build_match_flag.resize(build_match_flags.size(), 0);
+    }
+    DCHECK_EQ(build_match_flags.size(), _shared_build_match_flag.size());
+    vectorized::ColumnHelper::or_two_filters(&_shared_build_match_flag, build_match_flags.data());
+
+    return is_last;
+}
+
+const std::vector<uint8_t> CrossJoinContext::get_shared_build_match_flag() const {
+    DCHECK_EQ(_num_post_probers, _num_left_probers) << "all probers should share their states";
+    return _shared_build_match_flag;
+}
+
+void CrossJoinContext::append_build_chunk(int32_t sinker_id, vectorized::ChunkPtr chunk) {
+    _tmp_chunks[sinker_id].push_back(chunk);
```

Comment:
_tmp_chunks or build chunks?

Reply:
- `_tmp_chunks` is for each building operator, same as input data
- `_build_chunks` would merge build chunks into `4096` rows to avoid fragmented rows.

I could write some comments for understanding.